ecssd_agent.go

```diff
@@ -371,13 +408,17 @@ func main() {
 	}
 	configuration.HostedZoneId = zoneId
 	metadataClient := ec2metadata.New(session.New())
-	hostname, err := metadataClient.GetMetadata("/hostname")
-	configuration.Hostname = hostname
+	localIP, err := metadataClient.GetMetadata("/local-ipv4")
+	configuration.Hostname = "ip-" + strings.Replace(localIP, ".", "-", 3) + "." + DNSName
```

Comment:
Ingmar, the main reason I decided to force the records to point to "ip-a-b-d-c.domain" is for stale record cleanup. Let's say for example that someone uses the format "host1.domain.tld, host2.domain.tld, etc....". Lets also assume that someone has A records in the private hosted zone that were not created by the ecssd_agent, www.domain.tld, and hosted.domain.tld. Without forcing the srv records and automatically created A records to a known format, I cannot cleanup stale A records without risk of deleting non ecssd_agent created records.